subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -6,24 +6,39 @@
 
 #include <zephyr/devicetree.h>
 #include <zephyr/kernel.h>
-#include <string.h>
-#include <zephyr/mgmt/ec_host_cmd/backend.h>
 #include <zephyr/mgmt/ec_host_cmd/ec_host_cmd.h>
-
-BUILD_ASSERT(DT_HAS_CHOSEN(zephyr_ec_host_interface),
-	"Must choose zephyr,ec-host-interface in device tree");
-
-#define DT_HOST_CMD_DEV DT_CHOSEN(zephyr_ec_host_interface)
+#include <zephyr/mgmt/ec_host_cmd/backend.h>
+#include <string.h>
 
 #define RX_HEADER_SIZE (sizeof(struct ec_host_cmd_request_header))
 #define TX_HEADER_SIZE (sizeof(struct ec_host_cmd_response_header))
 
-/**
- * Used by host command handlers for their response before going over wire.
- * Host commands handlers will cast this to respective response structures that may have fields of
- * uint32_t or uint64_t, so this buffer must be aligned to protect against the unaligned access.
- */
-static uint8_t tx_buffer[CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER] __aligned(8);
+#define EC_HOST_CMD_DEFINE(_name)                                                                  \
+	COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_DEF,                                      \
+		    (static uint8_t _name##_rx_buffer[CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER];), ()) \
+	COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,                                      \
+		    (static uint8_t _name##_tx_buffer[CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER];), ()) \
+	static K_KERNEL_STACK_DEFINE(_name##stack, CONFIG_EC_HOST_CMD_HANDLER_STACK_SIZE);         \
+	static struct k_thread _name##thread;                                                      \
+	static struct ec_host_cmd _name = {                                                        \
+		.rx_ctx =                                                                          \
+			{                                                                          \
```

Comment:
keep the brace inline? same below

Reply:
I believe Fabio's comment still applies.

Reply:
I know, but I think a custom format is not required here.

Reply:
Well, your name is the maintainer one for this after this PR so your call I guess, but I think this is just plain wrong compared the the coding style. I'd say shut clang off and indent manually.